components/TokenManagePage/Media.tsx

```diff
@@ -0,0 +1,92 @@
+"use client";
+
+import { useTokenProvider } from "@/providers/TokenProvider";
+import { useState, useEffect } from "react";
+import { Input } from "@/components/ui/input";
+import { Textarea } from "@/components/ui/textarea";
+
+const Media = () => {
+  const { metadata } = useTokenProvider();
+  const { data: meta, isLoading } = metadata;
+
+  const [formData, setFormData] = useState({
+    title: "",
+    description: "",
+    image: "",
+    animationUrl: "",
+  });
+
+  useEffect(() => {
+    if (meta) {
+      setFormData({
+        title: meta.name || "",
+        description: meta.description || "",
+        image: meta.image || "",
+        animationUrl: meta.animation_url || "",
+      });
+    }
+  }, [meta]);
+
+  if (isLoading || !meta) {
+    return (
+      <div className="px-4 md:px-10 w-full pt-4">
+        <div className="animate-pulse">
+          <div className="h-4 bg-grey-moss-200 rounded w-1/4 mb-4"></div>
+          <div className="h-64 bg-grey-moss-200 rounded mb-4"></div>
+          <div className="space-y-2">
+            <div className="h-4 bg-grey-moss-200 rounded"></div>
+            <div className="h-4 bg-grey-moss-200 rounded w-3/4"></div>
+          </div>
+        </div>
+      </div>
```

Comment:
Single Responsibility Principle
- actual: `skeleton` component defined within the same file as the loaded component
- required: new component file for `MediaSkeleton`